localstack-core/localstack/services/sns/provider.py

```diff
@@ -1035,6 +1045,29 @@ def get_next_page_token_from_arn(resource_arn: str) -> str:
     return to_str(base64.b64encode(to_bytes(resource_arn)))
 
 
+def _get_byte_size(payload: str | bytes) -> int:
+    # Calculate the real length of the byte object if the object is a string
+    return len(to_bytes(payload))
+
+
+def get_total_publish_size(
+    message_body: str, message_attributes: MessageAttributeMap | None
+) -> int:
+    size = len(to_bytes(message_body))
```

Comment:
we can use the helper function `_get_byte_size` to calculate the message body size here.

Reply:
Ah right, oopsie 😄 good catch!